source/libNsdl/src/sn_grs.c

```diff
@@ -23,7 +23,7 @@
  */
 #include <string.h>
 #include <stdlib.h>
-
+#include <assert.h>
```

Comment:
Actually, some sort of tr_assert() extension to the trace library might make sense. You do really want the output going to the trace location, so that's presumably what a default ns_types.h implementation would do, in which case maybe it should just be in the trace to start with?

Don't forget the trace location isn't necessarily just a UART - it could be some sort of internal log-storage-for-later-extraction system. assert()'s default behaviour of "write to stderr" would not be helpful in that scenario.

Up to now, I've usually manually done tr_error() and return myself.